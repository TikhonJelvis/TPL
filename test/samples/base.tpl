print (precedenceOf (+))
print (precedenceOf (*))

a ~ b := a + b
precedence (~) 10
print (precedenceOf (~))

print (precedenceOf "*application*")

print (flip (-) 1 2)
-- TODO: test unary flip!

print (trimPrecs (exprToString $(+)))

super := {x : 10}
sub   := {y : 11} :> super

print (super.x)
print (sub.y)
print (sub.x)

super.y := 100
print (super.y)
print (sub.y)

super.z := 42
print (super.z)
print (sub.z)

super.y <- 0
print (super.y)
print (sub.y)

super.z <- 123
print (super.z)
print (sub.z)

-- TODO: Figure out how to test missing variables and the like!

print (if true 1 else (print "fail"))
print (if false (print "fail") else 1)

print (force $10)