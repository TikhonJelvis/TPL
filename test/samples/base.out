7
8
10
12
1
+
10
11
10
100
11
42
42
0
11
123
123
1
1
10
null
